### PR TITLE
Added document writing commands

### PR DESCRIPTION
**Pull request by suvadeet**
## Description of new Instant Answer, or changes

Added document writing commands

## Related Issues and Discussions
Fixing issue form Latex #3994 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable (for "More at {source_name}" link)
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [x] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [x] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [x] There is no unnecessary files in place (such as editor config files)
- [x] There is no API keys / secrets present
- [x] Tests are passing (run `$ duckpan test <fathead_id>`)
    - Tester should report any failures

**Ready to merge?**

- [x] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [x] The number of reviews is appropriate for this type of PR
- [x] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/latex_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/latex.json [hunk 1]
```diff
@@ -380,6 +386,154 @@
...
+            {
+                "key": "report",
+                "val": "Longer reports with several chapters, short books, thesis"
+            },
+            {
+                "key": "book",
+                "val": "Full length books"
+            },
+            {
+                "key": "slides",
```
Comment (nalinbhardwaj) [inline share/goodie/cheat_sheets/json/latex.json]: Either `beamer` class should be added or this should be replaced by it as `slides` are not really useful

File: share/goodie/cheat_sheets/json/latex.json [hunk 2]
```diff
@@ -380,6 +386,154 @@
...
+                "val": "Right align text"
+            },
+            {
+                "key": "\\begin\\{center\\}<your text>\\end\\{center\\}",
+                "val": "Center align text"
+            } 
+        ],
+        "Lists": [
+            {
+                "key": "\\begin\\{itemize\\}",
```
Comment (nalinbhardwaj) [inline share/goodie/cheat_sheets/json/latex.json]: `\begin{itemize}` and `\begin{enumerate}` and similarly `end` should be merged into single command with mention of how `itemize` and `enumerate` refer to ordered/unordered lists to make section more compact and clear.

File: share/goodie/cheat_sheets/json/latex.json [hunk 3]
```diff
@@ -380,6 +386,154 @@
...
+                "val": "End ordered list"
+            },
+            {
+                "key": "\\item",
+                "val": "All list entry must be preceded by \\item"
+            }
+        ],
+        "Special Characters": [
+            {
+                "key": "`<your text>'",
```
Comment (nalinbhardwaj) [inline share/goodie/cheat_sheets/json/latex.json]: Not 100% sure about this, but [ShareLatex](https://www.sharelatex.com/learn/Typesetting_quotations#/Reference_guide) says they are language dependent. Not a very good idea to add them in such a case.

File: share/goodie/cheat_sheets/json/latex.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
     "id": "latex_cheat_sheet",
     "name": "Latex",
-    "description": "Latex math mode commands and syntax",
+    "description": "Latex math mode and document writing commands and syntax",
```
Comment (nalinbhardwaj) [inline share/goodie/cheat_sheets/json/latex.json]: Please add required aliases as well.

File: share/goodie/cheat_sheets/json/latex.json [hunk 5]
```diff
@@ -380,6 +386,154 @@
...
+            {
+                "key": "\\date\\{date\\}",
+                "val": "Date of document, Optional; Must go before \\begin{document}"
+            },
+            {
+                "key": "\\maketitle",
+                "val": "Produces title; Goes after \\begin{document}"
+            }
+        ],
+         "Document Formatting": [
```
Comment (nalinbhardwaj) [inline share/goodie/cheat_sheets/json/latex.json]: You should add information about commenting using `%` here.

**Comment by suvadeet** [comment]
I've updated the file per your request. Thanks

File: share/goodie/cheat_sheets/json/latex.json [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
+    "description": "Latex math mode and document writing commands and syntax",
 
     "metadata": {
         "sourceName": "Wikibooks",
         "sourceUrl" : "https://en.wikibooks.org/wiki/LaTeX"
     },
 
     "aliases": [
-        "tex"
+        "tex", "latex document"
```
Comment (nalinbhardwaj) [inline share/goodie/cheat_sheets/json/latex.json]: Please add triggers `latex doc`, `latex writing`, `latex document writing`, `latex doc writing`, `tex doc`, `tex writing`, `tex document writing`, `tex doc writing` as well.

File: share/goodie/cheat_sheets/json/latex.json [hunk 7]
```diff
@@ -380,6 +386,138 @@
...
+            {
+                "key": "\\begin\\{document\\}",
+                "val": "Starts a document"
+            },
+            {
+                "key": "\\end\\{document\\}",
+                "val": "Ends a document"
+            }
+        ],
+        "Document Classes": [
```
Comment (nalinbhardwaj) [inline share/goodie/cheat_sheets/json/latex.json]: As mentioned in previous review, please add class `beamer` as well. It is described [here](https://www.sharelatex.com/learn/Beamer).

File: share/goodie/cheat_sheets/json/latex.json [hunk 8]
```diff
@@ -380,6 +386,138 @@
...
+                "val": "Create unordered list. Use enumerate instead of itemize for ordered list."
+            },
+            {
+                "key": "\\item",
+                "val": "All list entry must be preceded by \\item"
+            }
+        ],
+        "Special Characters": [
+            {
+                "key": "``<your text>''",
```
Comment (nalinbhardwaj) [inline share/goodie/cheat_sheets/json/latex.json]: Remove double quotations from here as well. As mentioned in previous review, we don't want to include anything specific to english-us.

File: share/goodie/cheat_sheets/json/latex.json [hunk 9]
```diff
@@ -380,6 +386,138 @@
...
+                "val": "Right align text"
+            },
+            {
+                "key": "\\begin\\{center\\}<your text>\\end\\{center\\}",
+                "val": "Center align text"
+            } 
+        ],
+        "Lists": [
+            {
+                "key": "\\begin\\{itemize\\}<\\item your text>\\end\\{itemize\\}",
```
Comment (nalinbhardwaj) [inline share/goodie/cheat_sheets/json/latex.json]: This needs to be `\\begin\\{itemize\\}\\item <your text> \\end\\{itemize\\}` instead. That is, the bracket for **<** your text> should begin after `\item`

**Comment by suvadeet** [comment]
I've updated the file. Hopefully, I've caught everything. Thanks

**Comment by suvadeet** [comment]
Thanks for your guidance @nalinbhardwaj

**Pull request merged**